packages/nitro-protocol/contracts/examples/RockPaperScissors.sol

```diff
@@ -87,75 +98,146 @@ contract RockPaperScissors is ForceMoveApp {
     // action requirements
 
     function requireValidPROPOSE(
-        VariablePart memory a,
-        VariablePart memory b,
-        RPSData memory appDataA,
-        RPSData memory appDataB
-        ) private pure
-        outcomeUnchanged(a,b)
-        stakeUnchanged(appDataA, appDataB)
-        allocationsNotLessThanStake(a, b, appDataA, appDataB)
-        {
-        }
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    )
+        private
+        pure
+        outcomeUnchanged(fromPart, toPart)
+        stakeUnchanged(fromGameData, toGameData)
+        allocationsNotLessThanStake(fromPart, toPart, fromGameData, toGameData)
+    {}
 
     function requireValidREJECT(
-        VariablePart memory a,
-        VariablePart memory b,
-        RPSData memory appDataA,
-        RPSData memory appDataB
-        ) private pure  {
-            // TODO
-        }
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        // TODO
+    }
 
     function requireValidACCEPT(
-        VariablePart memory a,
-        VariablePart memory b,
-        RPSData memory appDataA,
-        RPSData memory appDataB
-        ) private pure
-        {
-            // TODO
-        }
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        // TODO
+    }
+
+    function requireValidREVEAL(
+        VariablePart memory fromPart,
+        VariablePart memory toPart,
+        RPSData memory fromGameData,
+        RPSData memory toGameData
+    ) private pure {
+        uint256 playerFirstWinnings;
+        uint256 playerSecondWinnings;
+        require(
+            toGameData.playerSecondWeapon == fromGameData.playerSecondWeapon,
+            "Player Second's weapon should be the same between commitments."
+        );
+
+        // check hash matches
+        // need to convert Weapon -> uint256 to get hash to work
+        bytes32 hashed = keccak256(
+            abi.encodePacked(uint256(toGameData.playerFirstWeapon), toGameData.salt)
+        );
+        require(hashed == fromGameData.preCommit, 'The hash needs to match the precommit');
+
+        // calculate winnings
+        (playerFirstWinnings, playerSecondWinnings) = winnings(
+            toGameData.playerFirstWeapon,
+            toGameData.playerSecondWeapon,
+            toGameData.stake
+        );
+
+        Outcome.OutcomeItem[] memory outcomeFrom = abi.decode(fromPart.outcome, (Outcome.OutcomeItem[]));
```

Comment:
`fromOutcome` instead of `outcomeFrom` and so on would match the way we modify names elsewhere in this contract.